python/ray/air/tests/test_errors.py

```diff
@@ -195,6 +200,73 @@ def test_driver_error_with_trainer(ray_start_4_cpus, tmp_path, error_on):
     assert TrainingFailedError._FAILURE_CONFIG_MSG not in str(exc_info.value)
 
 
+class FailingDataParallelTrainer(DataParallelTrainer):
+    def _propagate_results(self, training_results: List[_TrainingResult]):
+        super()._propagate_results(training_results)
+
+        if training_results[0].metrics["round"] == 2:
+            print("[Round 2b] Failing the coordinator actor!!")
+            # NOTE: ray.actor.exit_actor raises a SystemExit, which only exits out
+            # of this training thread. This is because the actors spawned by
+            # Ray Train looks like:
+            # Coordinator (event handling main thread + training thread <- we are here)
+            #   Worker 0 (event handling main thread + training thread)
+            #   ...
+            #   Worker n-1 (same as above)
+            # We need to use os._exit to exit out of the entire actor main thread.
+            os._exit(1)
+
+
+def test_preemption_error(tmp_path):
+    """This test simulates preemption actor errors at different levels of execution
+    and checks that the error counting is correct:
+    - Round 0: Actor error in the training worker. (shouldn't be counted)
+    - Round 1: User error in the training worker.
+    - Round 2: Actor error in the coordinator actor. (shouldn't be counted)
+    - Round 3: No error.
```

Comment:
Yeah good idea.

Reply:
I am not able to figure out how to mock a property on the `RayActorError` that core raises -- any ideas here?

I tried this:
```python
class MockRayActorError(ray.exceptions.RayActorError):
    preempted = True

monkeypatch.setattr(
    ray.tune.execution.tune_controller, "RayActorError", MockRayActorError
)
monkeypatch.setattr(ray.exceptions, "RayActorError", MockRayActorError)
```


I was planning on reworking this test to use the actual `gcs_client.drain_node` API to mock the preemption instead of mocking the attribute. ([example here](https://github.com/ray-project/ray/blob/master/python/ray/tests/test_draining.py#L45))

Reply:
@jjyao any ideas here?